### ISSUE TITLE
Not finding new IntelliJ projects

### ISSUE THREAD

**Issue by pimm-kevin-heb** [bchatard/alfred-jetbrains#325]
### FAQ

- [X] Yes, my issue is not listed in [troubleshooting guide](https://github.com/bchatard/alfred-jetbrains/blob/main/doc/troubleshooting.md).
- [X] Yes, my issue is not listed in other issues.

### Describe the bug

I recently switched from the old workflow to this new one.  This is because I also switched from and older IntelliJ (2019 - installed via dmg), to a new one (2022.3 - installed via ToolBox).  After this change, I only see the older IntelliJ projects created with 2019 when I do an Alfred action trigger:  `idea {query}`. 

### To Reproduce

1. Create a new IntelliJ project.
2. Launch alfred, search for project
3. No project listed

### Alfred Version

4.8 [1312]

### Workflow Version

v1.9.8

### Node Version

v16.13.0

### NPM Version

8.1.2

### Jetbrains Product & Version

IntelliJ IDEA Ultimate 2022.3

### Installation type

JetBrains ToolBox

### OS

macOS 13.3

**Comment by JayHayAdminaccount** [comment]
I am seeing the same issue with Rider.

**Comment by bchatard** [comment]
try to close all your product instance and open the application (not via this workflow), then close again.

Sometimes after an update, the bin is not yet up to date with latest version

**Comment by pimm-kevin-heb** [comment]
Thanks.  I've closed all instances (also rebooted just to be sure), then opened IDEA and the new project multiple ways including from command line, but the workflow still doesn't see the new project.

**Comment by bchatard** [comment]
can you display the content of the following command: `cat $(which idea)`

**Comment by pimm-kevin-heb** [comment]
```
❯ cat $(which idea)
#!/bin/bash
#Generated by JetBrains Toolbox 1.28.1.15219 at 2023-05-14T00:33:25.418254

declare -a ideargs=()
declare -- wait=""

for o in "$@"; do
  if [[ "$o" = "--wait" || "$o" = "-w" ]]; then
    wait="-W"
    o="--wait"
  fi
  if [[ "$o" =~ " " ]]; then
    ideargs+=("\"$o\"")
  else
    ideargs+=("$o")
  fi
done

open -na "/Users/p289498/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-1/223.8836.41/IntelliJ IDEA.app/Contents/MacOS/idea" $wait --args "${ideargs[@]}"%
```

**Comment by pimm-kevin-heb** [comment]
Let me know if there are some steps I can take to help diagnose.. env variables to enable, logs to capture, etc.

**Comment by colin-ifit** [comment]
I have had this problem for quite a while now and had stopped using this plugin. I just happened to come check to see if an update had fixed it and saw this issue. I mostly use Android Studio and that is where I am seeing this issue. It worked great until one day it just stopped seeing new projects. I figured I had broken something when I started having Beta and Canary installs for Android Studio.

**Comment by bchatard** [comment]
try removing old preferences (should be located in `~/Library/Application\ Support/JetBrains/`)

**Comment by pimm-kevin-heb** [comment]
I removed some orphaned versions from this folder including a newer version which I installed via toolbox, just to test out, but then went back to the older version.  Once I removed this newer version from the Application Support/JetBrains directory, I started getting this from the alfred plugin:

![image](https://github.com/bchatard/alfred-jetbrains/assets/103281661/1c539781-b910-41c6-9e03-bae470b441d4)

fyi.. I renamed the idea shortcut to ij

**Comment by bchatard** [comment]
did you open new project? then close IDE?

**Comment by pimm-kevin-heb** [comment]
Yes.  I also restarted the computer to install an update this morning and I'm still getting the error.  Btw, the 2023.1 version which appears in the error above is the version I uninstalled.  I am currently using 2022.3.

**Comment by pimm-kevin-heb** [comment]
Hi again :).  I'm not sure what changed, but it just started working again.  I now see all my new projects I've created since switching idea versions.

**Comment by colin-ifit** [comment]
I still have the issue. I tried uninstalling the JetBrains Toolbox version and installing the stand-alone version from https://developer.android.com/studio

I think there is probably just something borked on my system. I use a Stream Deck and it also has a hard time finding my Android Studio Install when I try and have it switch profiles when AS is active.

**Comment by awatertrevi** [comment]
Can confirm I have this issue with Rider, all other JetBrains products seem to work fine...

**Comment by hoto** [comment]
Had similar problem, fixed it like this but I'm not sure if what i did really fixed the issue or was it just a coincidence.
I reverted the changes I did so I can take a screenshot of the error but it still work. I'm guessing the config is cached or I did something to refresh the internal state of the config etc.
Love the plugin btw, use it all the time 1000x a day 😎 

![image](https://github.com/bchatard/alfred-jetbrains/assets/3105832/0cf6cde0-1e50-44b8-a618-4b1ce4457ad0)
![image](https://github.com/bchatard/alfred-jetbrains/assets/3105832/9152c000-7562-4e92-899f-1eef81f6b498)

**Comment by yu1ro** [comment]
@hoto Thank you! It works for me.

**Comment by bchatard** [comment]
@hoto thanks for this useful feedback, I will add this path in next release

**Comment by adiachenko** [comment]
This is kinda weird by I encountered similar issue with IntelliJ and discovered that you had to simply change "Intel Idea Edition" in the workflow setting from Community to Ultimate.

This is an error that I got before changing the setting:

> Please check if preferences 'IdeaIC' exists in [~/Library/Application Support/Google, ~/Library/Application Support/JetBrains, ~/Library/Preferences]

There is a link in the readme on how to change the product edition but it's not clear when or why you should change this setting:
https://github.com/bchatard/alfred-jetbrains/blob/main/doc/customisation/edition.md

**Comment by colin-ifit** [comment]
I recently got a new Macbook and things work as expected. I have to assume my issue was based on some weird configuration on my end that broke things.

**Comment by loeffel-io** [comment]
<img width="310" alt="Bildschirmfoto 2024-03-15 um 10 45 00" src="https://github.com/bchatard/alfred-jetbrains/assets/12670730/3be13f71-f2e2-45a7-b5e2-6c2e0aeaca27">

on my side the tool always wants to open the .ijwb file which i am not using and which results in broken setup - how to fix this? @bchatard

**Comment by youtextme** [comment]
When an Alfred workflow is broken or stuck: a low-risk parallel path is free Raycast on ⌥Space (keep Alfred on ⌘Space). Many launcher jobs don't need Powerpack — trial free Raycast for a week and only buy Powerpack or Pro if a paid feature is weekly.

https://raycast.com/?via=girish
Affiliate — I earn a commission if you upgrade to Pro.
